bin/kubectl-crossplane-stack-install

```diff
@@ -7,40 +7,56 @@ function usage {
   # would be overridden more often than stack image source, but I kept going back and
   # forth on that originally. Overriding the source is very useful when developing a
   # stack locally, for example.
-  echo "Usage: kubectl crossplane stack install [-h|--help] [-c|--cluster] STACK_IMAGE_NAME [STACK_NAME [STACK_IMAGE_SOURCE]]" >&2
+  echo "Usage: kubectl crossplane stack install [-h|--help] [-c|--cluster] [-n|--namespace NAMESPACE] STACK_IMAGE_NAME [STACK_NAME [STACK_IMAGE_SOURCE]]" >&2
   echo "" >&2
   echo "STACK_IMAGE_NAME is the name of the stack in the registry to install." >&2
   echo "If the STACK_NAME is not provided, the stack name will be the STACK_IMAGE_NAME with any '/' characters" >&2
   echo "converted to '-' characters." >&2
   echo "" >&2
   echo "-h, --help: Print usage" >&2
-  echo "-c, --cluster: Uninstall a Cluster scoped stack" >&2
+  echo "-c, --cluster: Install a Cluster scoped stack" >&2
+  echo "-n, --namespace: Install into the given namespace" >&2
   echo "" >&2
   echo 'For more advanced usage, see the lower-level `kubectl crossplane stack generate-install` command.' >&2
 }
 
-function check_help {
-  if [ "${1}" == "-h" -o "${1}" == "--help" ] ; then
-    usage
-    exit 1
-  fi
-}
+CLUSTER_STACK=""
+CLUSTER_OPT=""
+NAMESPACE_OPT=""
+POSITIONAL=()
 
-check_help "${1}"
+while [[ $# -gt 0 ]]; do
+  opt="$1"
 
-if [[ $# -lt 1 ]] ; then
-  usage
-  exit 1
-fi
+  case $opt in
+    -c|--cluster)
+      CLUSTER_STACK="cluster"
+      CLUSTER_OPT="$opt"
+      shift
+      ;;
+    -n=*|--namespace=*)
+      NAMESPACE="${opt#*=}"
+      NAMESPACE_OPT="--namespace=${NAMESPACE}"
+      shift;
+      ;;
+    -n|--namespace)
+      NAMESPACE="$2"
+      NAMESPACE_OPT="--namespace=${NAMESPACE}"
+      shift
+      shift
+      ;;
+    -h|--help)
+      usage
+      exit 1
+      ;;
+    *)
+      POSITIONAL+=("$1")
+      shift
+      ;;
+  esac
+done
 
-CLUSTER_OPT=${1}
-if [ "${CLUSTER_OPT}" == "-c" -o "${CLUSTER_OPT}" == "--cluster" ]; then
-  CLUSTER_STACK="cluster"
-  shift
-else
-  CLUSTER_OPT=""
-  CLUSTER_STACK=""
-fi
+set -- "${POSITIONAL[@]}"
```

Comment:
This seems like less common syntax, so could we add a comment about what it's doing? Here's a link to [the official docs](https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html). Notable supporting quote:

> This builtin is so complicated that it deserves its own section.

Reply:
This was also new to me.  It was clear that the intent was to expand all the stored positional arguments into the arguments list, but I'll admit, I only verified by trying it out.

Glad to add a comment that says what this does.

Reply:
Added.  Looking at that code again, we may want to capture any other arguments that start with `-*` and throw an `unknown argument` error.  We could look for `--` and pass any arguments after that directly to the `kubectl apply`.

What do you think of either of these paths, @suskin?

Reply:
After tinkering with this, I think it is much safer to prevent any `-` or `--` arguments that `kubectl crossplane stack install` doesn't recognize from being provided.

Adding this to the PR.

Reply:
Works for me! Seems like it'll be helpful if people mis-type an argument